src/masking/mod.rs

```diff
@@ -1 +1,94 @@
+use core::mem::MaybeUninit;
+use core::num::{NonZeroUsize, Wrapping};
+
+use crate::capacity::MaskingCapacity;
+use crate::storage::{PartialStorage, Storage};
+
 mod tests;
+
+pub struct MaskingRingBuffer<
+    Item,
+    S: PartialStorage<Item = MaybeUninit<Item>, Capacity = MaskingCapacity>,
```

Comment:
It would also be possible to take any `S` such that `S::Capacity: Into<MaskingCapacity>`; this would let us support `PowerOfTwoCapacity`.

Reply:
We can't store the into'ed capacity though

Reply:
Unless you want to add a field, I suppose

Reply:
No, the point is that `Storage` manages the capacity; if the `Into` implementation is doing actual work, then that's just a bad choice of capacity.  This is equivalent to the older ring buffer implementations where you need to compute the mask as `capacity - 1` every time.

Reply:
Yeah that's what I thought, but then the current implementation forces the fastest behaviour

Reply:
After thinking about it some more, I agree; there aren't going to be many use cases where an existing storage using an incorrect capacity type needs to be provided to a ring buffer.  Closing.